src/coreclr/src/gc/gc.cpp

```diff
@@ -20722,61 +20663,46 @@ void gc_heap::process_mark_overflow_internal (int condemned_gen_number,
 
         dprintf(3,("Processing Mark overflow [%Ix %Ix]", (size_t)min_add, (size_t)max_add));
 #ifdef MULTIPLE_HEAPS
-            for (int hi = 0; hi < n_heaps; hi++)
-            {
-                gc_heap*  hp = g_heaps [(heap_number + hi) % n_heaps];
+    for (int hi = 0; hi < n_heaps; hi++)
+    {
+        gc_heap*  hp = g_heaps [(heap_number + hi) % n_heaps];
 
 #else
-            {
-                gc_heap*  hp = 0;
+    {
+        gc_heap*  hp = 0;
 
 #endif //MULTIPLE_HEAPS
-        BOOL small_object_segments = TRUE;
-        int align_const = get_alignment_constant (small_object_segments);
-        generation* gen = hp->generation_of (condemned_gen_number);
-        heap_segment* seg = heap_segment_in_range (generation_start_segment (gen));
+        int align_const = get_alignment_constant (TRUE);
+        int up_to_generation = full_p ? total_generation_count : condemned_gen_number + 1;
```

Comment:
it is not exactly count, since we do not start from 0. I will make it `limit`

---
In reply to: [362296750](https://github.com/dotnet/runtime/pull/235#discussion_r362296750) [](ancestors = 362296750)